aws/resource_aws_emrcontainers_virtual_cluster_test.go

```diff
@@ -0,0 +1,467 @@
+					resource.TestCheckResourceAttr(resourceName, "container_provider.0.info.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "container_provider.0.info.0.eks_info.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "container_provider.0.info.0.eks_info.0.namespace", "default"),
+					resource.TestCheckResourceAttr(resourceName, "container_provider.0.type", "EKS"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "state", "RUNNING"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsEMRContainersVirtualCluster_disappears(t *testing.T) {
+	var vc emrcontainers.DescribeVirtualClusterOutput
+
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_emrcontainers_virtual_cluster.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsEMRContainersVirtualClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsEMRContainersVirtualClusterBasicConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsEMRContainersVirtualClusterExists(resourceName, &vc),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsEMRContainersVirtualCluster(), resourceName),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsEMRContainersVirtualClusterExists(resourceName string, vc *emrcontainers.DescribeVirtualClusterOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No EMR containers virtual cluster ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).emrcontainersconn
+		resp, err := conn.DescribeVirtualCluster(&emrcontainers.DescribeVirtualClusterInput{
+			Id: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return fmt.Errorf("Error getting EMR containers virtual cluster: %s", err.Error())
+		}
+
+		*vc = *resp
+
+		return nil
+	}
+}
+
+func testAccCheckAwsEMRContainersVirtualClusterDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).emrcontainersconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_emrcontainers_virtual_cluster" {
+			continue
+		}
+
+		_, err := conn.DescribeVirtualCluster(&emrcontainers.DescribeVirtualClusterInput{
+			Id: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if isAWSErr(err, emrcontainers.ErrCodeResourceNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		return fmt.Errorf("Expected EMR containers virtual cluster to be destroyed, %s found", rs.Primary.ID)
```

Comment:
EMR virtual clusters don't disappear when you delete them. They will still respond to a `describe-virtual-cluster` request, but return a `TERMINATED` state.

This method should check if the cluster is terminated instead of if we get a `ResourceNotFoundException`.

```suggestion
		resp, err := conn.DescribeVirtualCluster(&emrcontainers.DescribeVirtualClusterInput{
			Id: aws.String(rs.Primary.ID),
		})

		if err != nil {
			if isAWSErr(err, emrcontainers.ErrCodeResourceNotFoundException, "") {
				return nil
			}
			return err
		}

		if resp.VirtualCluster.State != nil {
			clusterState := *(resp.VirtualCluster.State)
			if clusterState != "TERMINATED" {
				return fmt.Errorf("Expected EMR containers virtual cluster to be destroyed, %s found", rs.Primary.ID)
			}
		}
```

Reply:
Hey @dacort!

Thanks for the suggestion :) Turns out I had forgotten to push one of the last commits I made, which was specifically around that part. I've just pushed it now so feel free to take another look :)